Submitted By: Pierre Cyr <pierre.cyr@mail.mcgill.ca>
Date: 2004-05-18
Initial Package Version: 1.4.3
Upstream Status: Submitted to LFS and SVGAlib developers 
Origin: http://patches.linuxfromscratch.org (includes svgalib-1.4.3-gcc3-1.patch)
        and myself
Description: This patch allows to compile svgalib-1.3.4 using gcc-3.4 and its
             more strict syntax.

$LastChangedBy$
$Date$

diff -Naur svgalib-1.4.3.orig/src/mach32.c svgalib-1.4.3/src/mach32.c
--- svgalib-1.4.3.orig/src/mach32.c	1999-09-29 16:52:14.000000000 +0000
+++ svgalib-1.4.3/src/mach32.c	2004-05-19 02:11:16.148248344 +0000
@@ -1544,7 +1544,7 @@
 		goto writerr;
 	    printf("mach32: Notice: new EEPROM file >%s< succesful created.\n",
 		   eeprom_fname);
-	  finish_w_eeprom:
+	  finish_w_eeprom: fd == NULL; //Useless line to compile
 	}
 	/* Change eeprom contents if requested: */
 	if (!(eeprom_option & EEPROM_USE_MEMCFG))
diff -Naur svgalib-1.4.3.orig/src/s3.c svgalib-1.4.3/src/s3.c
--- svgalib-1.4.3.orig/src/s3.c	2000-05-03 12:23:57.000000000 +0000
+++ svgalib-1.4.3/src/s3.c	2004-05-19 02:11:41.643372496 +0000
@@ -474,7 +474,6 @@
 	    dac_used->restoreState(regs);
 	    return old_values;
 	}
-    default:
     }
     return 0;
 }
diff -Naur svgalib-1.4.3.orig/src/vga.c svgalib-1.4.3/src/vga.c
--- svgalib-1.4.3.orig/src/vga.c	2001-05-29 14:37:55.000000000 +0000
+++ svgalib-1.4.3/src/vga.c	2004-05-19 02:09:10.637328904 +0000
@@ -3914,7 +3914,7 @@
 
 #define ML_GETINT(x) \
 	ptr = strtok(NULL, " "); if(!ptr) break; \
-	mmt.##x = atoi(ptr);
+	mmt.x = atoi(ptr);
 
 	    ML_GETINT(HDisplay);
 	    ML_GETINT(HSyncStart);
